pkg/controller/job/job_controller.go

```diff
@@ -494,8 +496,12 @@ func (jm *JobController) syncJob(key string) (bool, error) {
 	var failureReason string
 	var failureMessage string
 
-	jobHaveNewFailure := failed > job.Status.Failed
-	exceedsBackoffLimit := jobHaveNewFailure && (int32(previousRetry)+1 > *job.Spec.BackoffLimit)
+	jobHaveNewFailure := (failed > job.Status.Failed)
+	// new failures happen when status does not reflect the failures and active
+	// is different than parallelism, otherwise the previous controller loop
+	// failed updating status so even if we pick up failure it is not a new one
+	exceedsBackoffLimit := jobHaveNewFailure && (active != *job.Spec.Parallelism) &&
```

Comment:
Would you explain why `active != *job.Spec.Parallelism` is needed?

Reply:
This is for the case when a controller failed an update, but went through the whole process, which means that the number of active is exactly what it should be. There's no other way right now to figure that the previous loop failed.